lib/src/session.dart

```diff
@@ -229,9 +236,19 @@ class Session with ChangeNotifier {
       print('📗 session._jscOnUnreadsChange: $unreadsJson');
     }
 
-    unreads?.onChange?.call(
-      unreadsJson.map((unread) => UnreadConversation.fromJson(unread)).toList(),
-    );
+    if (onUnreadsChange != null) {
+      onUnreadsChange!.call(
+        unreadsJson
+            .map((unread) => UnreadConversation.fromJson(unread))
+            .toList(),
+      );
+    } else if ((unreads != null) && (unreads!.onChange != null)) {
+      unreads!.onChange!.call(
```

Comment:
Side note: I hate how dart can't see we have a null check just before this and we don't need the ! here.

Reply:
Agreed, Typescript demonstrated that it's possible to infer that.